tests/regression/test_fdm.py

```diff
@@ -0,0 +1,200 @@
+        family = "RTCF" if ndim == 2 else "NCF"
+        return FunctionSpace(mesh, family, degree)
+    else:
+        if ndim == 1:
+            family = "DG" if element == "dg" else "CG"
+        else:
+            family = "DQ" if element == "dg" else "Q"
+        return VectorFunctionSpace(mesh, family, degree)
+
+
+@pytest.mark.skipcomplex
+def test_direct_solver(fs):
+    mesh = fs.mesh()
+    x = SpatialCoordinate(mesh)
+    uex = dot(x, x)
+    ncomp = fs.ufl_element().value_size()
+    if ncomp > 1:
+        uex = as_vector([uex + Constant(k) for k in range(ncomp)])
+
+    n = FacetNormal(mesh)
+    Fex = grad(uex)
+    B = -div(Fex)
+    T = dot(Fex, n)
+    uh = Function(fs)
+    u = TrialFunction(fs)
+    v = TestFunction(fs)
+
+    subs = (1, 3)
+    if mesh.layers:
+        subs += ("top",)
+
+    bcs = [DirichletBC(fs, uex, sub) for sub in subs]
+
+    sub_Dir = "everywhere" if "on_boundary" in subs else tuple(s for s in subs if type(s) == int)
+    if sub_Dir == "everywhere":
+        sub_Neu = ()
+    else:
+        sub_Neu = tuple(set(mesh.exterior_facets.unique_markers) - set(s for s in subs if type(s) == int))
+
+    if mesh.layers:
+        dS_int = dS_v + dS_h
+        ds_Dir = ds_v(sub_Dir)
+        ds_Neu = ds_v(sub_Neu)
+        if "bottom" in subs:
+            ds_Dir += ds_b
+        else:
+            ds_Neu += ds_b
+        if "top" in subs:
+            ds_Dir += ds_t
+        else:
+            ds_Neu += ds_t
+    else:
+        dS_int = dS
+        ds_Dir = ds(sub_Dir)
+        ds_Neu = ds(sub_Neu)
+
+    N = fs.ufl_element().degree()
+    try:
+        N, = set(N)
+    except TypeError:
+        pass
+
+    eta = Constant((N+1)**2)
+    h = CellVolume(mesh)/FacetArea(mesh)
+    penalty = eta/h
+
+    a = (inner(grad(v), grad(u))*dx
+         + inner(outer_jump(v, n), avg(penalty) * outer_jump(u, n)) * dS_int
+         - inner(avg(grad(v)), outer_jump(u, n)) * dS_int
+         - inner(avg(grad(u)), outer_jump(v, n)) * dS_int
+         + inner(v, penalty * u) * ds_Dir
+         - inner(grad(v), outer(u, n)) * ds_Dir
+         - inner(grad(u), outer(v, n)) * ds_Dir)
+
+    L = (inner(v, B)*dx
+         + inner(v, T)*ds_Neu
+         + inner(v, penalty * uex) * ds_Dir
+         - inner(grad(v), outer(uex, n)) * ds_Dir)
+
+    problem = LinearVariationalProblem(a, L, uh, bcs=bcs)
+    solver = LinearVariationalSolver(problem, solver_parameters={
+        "mat_type": "matfree",
+        "ksp_type": "cg",
+        "ksp_atol": 0.0E0,
+        "ksp_rtol": 1.0E-8,
+        "ksp_max_it": 20,
+        "ksp_monitor": None,
+        "ksp_norm_type": "unpreconditioned",
+        "pc_type": "python",
+        "pc_python_type": "firedrake.FDMPC",
+        "fdm_pc_type": "cholesky",
+        "fdm_pc_factor_mat_solver_type": "mumps",
+        "fdm_pc_factor_mat_ordering_type": "nd",
+    }, appctx={"eta": eta, })
+    solver.solve()
+
+    assert solver.snes.ksp.getIterationNumber() == 1 and norm(uex-uh, "H1") < 1.0E-8
```

Comment:
I don't see any tests that pass `mu` or `beta`. Can you add a test that exercises this functionality? (Not least so that there's an example of how to use it!)